### PR TITLE
Saving time and resources when list reload in Library

### LINKED ISSUE #538: App crashes when changing ZIM language in landscape mode
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request

### Step 1: Are you ready to report?
   - Your issue may already be reported! Please search on the [issue track](../) before creating one.
   - [x] I have tested with the last (dev) version of the software.  ( If not, and if possible, run your test again with the most recent version available.)

### Step 2: What is your environment?
  * version of the software you use :  Commit 149088e
  * device / operating system : Android (Nexus 4)
  
### Step 3: Describe the problem:

  * What do you obtain? A copy of the error message or a screenshot is always useful.
screenshot_2018-03-01-14-35-49

  * When does this occur? 
In Landscape view, on the "Get Content" page under library, clicking the "globe" icon crashes the app

#### Steps to reproduce:

  1. Rotate device to be *portrait*
  2. Go to "Get Content"
  3. Rotate device to be landscape
  4. Click "Online"
  5. Click the globe icon
  
  * What should be the behaviour from your point of view? How do you expect the service to work?

### Step 4 (optional) : Other Comments:
  * What is the context of this ticket? If not obvious, explain why you need to do this.
 
  * If you have an idea about the technical background of the ticket, please share it.
  
  
#### Feature Description

Please describe the feature you want to add to the project.

### PR DESCRIPTION
**Pull request by amerinoo**
Fixes #538 @GeVic comment.

Changes: [Add here what changes were made in this issue and if possible provide links.]
Use Fragment methods onSaveInstanceState and onViewStateRestored to reload the books. Now the app don't use Internet every time the user rotates the device. As you can see on the GIF the reload is quite faster.
Screenshots/GIF for the change: [If possible, please add relevant screenshots/GIF]

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java [hunk 1]
```diff
@@ -153,16 +156,11 @@ public void showBooks(LinkedList<Book> books) {
       displayNoItemsAvailable();
       return;
     }
-
+    this.books=books;
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java]: spacing

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java [hunk 2]
```diff
@@ -85,7 +88,7 @@
   @BindView(R.id.library_swiperefresh)
   SwipeRefreshLayout swipeRefreshLayout;
 
-  private ArrayList<Book> books = new ArrayList<>();
+  private LinkedList<Book> books;
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java]: Can we change this to a generic List? Why has it been changed here?

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java [hunk 3]
```diff
@@ -153,16 +156,11 @@ public void showBooks(LinkedList<Book> books) {
...
     Log.i("kiwix-showBooks", "Contains:" + books.size());
     libraryAdapter.setAllBooks(books);
-    if (faActivity.searchView != null) {
-      libraryAdapter.getFilter().filter(
-          faActivity.searchView.getQuery(),
-          i -> stopScanningContent());
-    } else {
-      libraryAdapter.getFilter().filter("", i -> stopScanningContent());
-    }
+    String query = (faActivity.searchView != null)?faActivity.searchView.getQuery().toString():"";
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.java]: Please refactor this. Ternaries are not amazing and this is bad style as you could use == null and invert. Spacing as well.

**Comment by prajurock** [comment]
@abdulwd @soloturn @mhutti1  when I rotate app then its start fetching content again as I said in #958

**Comment by mhutti1** [comment]
Yes I reverted the change as it causes issues on some Samsung devices. Coudn't work out if it was tracked by an issue but reopened it now.

**Pull request merged**